pallets/hyperdrive-outgoing/src/lib.rs

```diff
@@ -311,9 +364,12 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
     /// This function should be combined with a check (not included!) if there was at least one new message to snapshot.
     fn maximum_blocks_before_snapshot_reached(current_block: T::BlockNumber) -> bool {
         // check if we should create new snapshot
-        let (last_block, _last_message_excl): (T::BlockNumber, LeafIndex) =
-            Self::snapshot_meta(Self::next_snapshot_number().saturating_sub(1))
-                .unwrap_or((0u32.into(), 0));
+        let (_root_hash, last_block, _last_message_excl): (_, T::BlockNumber, LeafIndex) =
+            Self::snapshot_meta(Self::next_snapshot_number().saturating_sub(1)).unwrap_or((
+                Default::default(),
+                0u32.into(),
+                0,
+            ));
 
         current_block.saturating_sub(last_block) >= T::MaximumBlocksBeforeSnapshot::get().into()
```

Comment:
We should have into consideration the first snapshot, right?
```suggestion
        0u32.ne(&last_block) && current_block.saturating_sub(last_block) >= T::MaximumBlocksBeforeSnapshot::get().into()
```

Reply:
fixed it properly respecting`first_mmr_block_number()`, now we don't create an empty initial snapshot when hyperdrive outgoing gets installed (by runtime upgrade) and also wait `MaximumBlocksBeforeSnapshot` before creating the first one.